src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,460 @@
+            getRootProps,
+            getToggleButtonProps,
+            isOpen,
+            highlightedIndex,
+            selectItemAtIndex,
+          }) => {
+            return (
+              <div style={styles.containerDiv} onClick={this.onContainerClick.bind(this, isOpen)}>
+                <span aria-live="assertive" style={styles.ariaLiveSpan}>
+                  {this.state.message}
+                </span>
+                {multiple && this.renderActiveValues(styles)}
+                {search &&
+                  this.renderInput(
+                    styles,
+                    variables,
+                    getRootProps,
+                    getInputProps,
+                    highlightedIndex,
+                    selectItemAtIndex,
+                  )}
+                {toggleButton && this.renderToggleButton(getToggleButtonProps, styles, isOpen)}
+                {this.renderList(
+                  styles,
+                  variables,
+                  getMenuProps,
+                  getItemProps,
+                  isOpen,
+                  highlightedIndex,
+                )}
+              </div>
+            )
+          }}
+        </Downshift>
+      </ElementType>
+    )
+  }
+
+  private renderInput(
+    styles,
+    variables,
+    getRootProps,
+    getInputProps,
+    highlightedIndex,
+    selectItemAtIndex,
+  ): JSX.Element {
+    const { multiple, placeholder } = this.props
+    const { searchQuery, value } = this.state
+
+    return (
+      <Input
+        inputRef={input => (this.inputRef = input)}
+        onFocus={this.onInputFocus}
+        onKeyUp={multiple && this.onInputKeyUpIfMultiple}
+        styles={styles.editTextDiv}
+        wrapper={{ ...getRootProps({ refKey: 'slotRef' }, { suppressRefError: true }) }}
+        variables={{ inputFocusBorderColor: variables.editTextInputFocusBorderColor }}
+        input={{
+          type: 'text',
+          styles: styles.editTextInput,
+          placeholder:
+            searchQuery.length > 0 || (multiple && (value as DropdownListItem[]).length > 0)
+              ? ''
+              : placeholder,
+          ...getInputProps({
+            onBlur: this.onInputBlur,
+            onKeyDown: this.onInputKeyDown.bind(this, highlightedIndex, selectItemAtIndex),
+          }),
+        }}
+      />
+    )
+  }
+
+  private renderToggleButton(getToggleButtonProps, styles, isOpen) {
+    return (
+      <Icon
+        name={`chevron ${isOpen ? 'up' : 'down'}`}
+        as="button"
+        tabIndex="-1"
+        styles={styles.toggleButton}
+        {...getToggleButtonProps()}
+      />
+    )
+  }
+
+  private renderList(styles, variables, getMenuProps, getItemProps, isOpen, highlightedIndex) {
+    return (
+      <List
+        {...getMenuProps({ refKey: 'listRef' })}
+        styles={styles.list}
+        aria-hidden={!isOpen}
+        items={isOpen ? this.renderItems(variables, getItemProps, highlightedIndex) : []}
+      />
+    )
+  }
+
```

Comment:
The thing to consider: this is a lot of code for one component, which brings complexity for maintaining it in the future. 
What about if we can break this component into small parts, the same way as we have ```Menu``` and ```MenuItems``` for example.
We can have next set of drop-down components, like ```DropdownList```, ```DropdownInput``` etc.
It's very easy to get lost now, so it can be even harder in the future.

Reply:
it is a point to consider, as I know that Semantic does something similar.

Right now the code is grouped into separate functions, so it's not that bad for reading it. We will follow up on this, thanks!

@jurokapsiar